developing/languages/GO.md

```diff
@@ -4,9 +4,184 @@
 * [Go Track on Exercism](https://exercism.io/tracks/go)
+
+## Testing
+
+### General
+
+* Use table-driven tests where appropriate.
+* Make judicious use of helper functions so that the intent of a test is not lost in a sea of error checking and boilerplate.
+  Use [`t.Helper()`](https://golang.org/pkg/testing/#T.Helper) in your test helper functions to keep stack traces clean.
+
+### Coverage
+
+* Always test exported functions.
+  Exported functions should be treated as an API layer for other packages.
+  Cover the expected behavior and error scenarios as a user of that API.
+* Try not to test unexported functions.
+  Unexported functions are implementation details of exported ones
+  and should not change the intended usage.
+  If you find that an unexported function is complex and needs testing,
+  it might mean it needs to be refactored as it's exported function elsewhere.
+
+## Packages
+
+### Prefer Standard Libraries
+
+In general,
+when selecting new packages,
+highly consider standard libraries over third party dependencies.
+One of the strengths of Go
+is its core packages,
+such as
+[http](https://golang.org/pkg/net/http/),
+[json](https://golang.org/pkg/encoding/json/),
+and [sql](https://golang.org/pkg/database/sql/).
+These libraries also use vocabulary and patterns
+easily accessible via popular public Go resources,
+which are often translatable to modern programming approaches
+in their respective areas.
+This creates an easier bridge
+for Engineers new to Go
+or domain areas (such as relational databases)
+to adjust and onboard.
+
+Since the third party ecosystem is still new,
+many packages tend to be immature with little community support,
+follow opinionated patterns inconsistent with Go idioms,
+or lack long term support.
+Be aware that choosing a new package
+is likely to involve contributing to its upkeep.
+
+## Time
+
+### Clock Dependency
+
+`time.Now()` can cause a lot of side effects in a codebase.
+One example is
+that you can't test the "current" time
+that happened in a function you called in the past
+
+For example, let's say we have the following:
+
+```go
+package mypackage
+
+import "time"
+
+func MyTimeFunc() time.Time {
+    return time.Now()
+}
+
+func TestMyTimeFunc(t *testing.T) {
+    if MyTimeFunc() != time.Now() {
+        // This will error!
+        // The time in the function and the test happen at different times
+        t.Errorf("time was not now")
+  }
+}
+```
+
+How do we test the contents of the return here?
+If we want to assert the time
+we need a way to know what `time.Now()` was when the function was called.
+
+Instead of directly using the `time` package,
+we can pass a clock as a dependency and call `.Now()` on that.
+Then in our tests, we can assert against that clock!
+The clock can be anything as long as it adheres to the `clock.Clock` interface
+as defined in the
+[facebookgo clock package](https://godoc.org/github.com/facebookgo/clock#Clock).
```

Comment:
Why not just pass in a `time.Time` value for now instead of a clock instance?

Reply:
I'm not sure I understand the question. `time.Time` is an instance of time, `clock` reads time, which is most commonly used to call `.Now`.